api/app/Models/User.php

```diff
@@ -618,18 +668,17 @@ public static function scopeEquity(Builder $query, ?array $equity): Builder
         return $query;
     }
 
-    public static function scopeGeneralSearch(Builder $query, ?string $search): Builder
+    public static function scopeGeneralSearch(Builder $query, ?array $searchTerms): Builder
     {
-        if ($search) {
-            $query->where(function ($query) use ($search) {
-                self::scopeName($query, $search);
-                $query->orWhere(function ($query) use ($search) {
-                    self::scopeEmail($query, $search);
-                });
-                $query->orWhere(function ($query) use ($search) {
-                    self::scopeTelephone($query, $search);
-                });
-            });
+        if ($searchTerms && is_array($searchTerms)) {
+            $combinedSearchTerm = implode('&', array_map('trim', $searchTerms));
+            $results = self::search($combinedSearchTerm)->usingWebSearchQuery()->get();
+
+            // Extract unique IDs from the search results
+            $uniqueIds = $results->pluck('id')->unique()->toArray();
+
+            // Use Eloquent builder to filter results based on unique IDs
+            $query->whereIn('id', $uniqueIds);
```

Comment:
Hmm, this isn't really covered by the issue, I think, but we should probably consider limiting the list of search results that come back.  No point in pulling up 1,000 hits if the user is going to look at the top, say, 20 hits.

Reply:
Oracle has a limit of 1000 terms that can be passed to an `in` clause.  I don't think Postgres has that limit but it would still probably burn a huge amount of ram to do that here.

Reply:
I changed it to pick up only top 100 to get this going.
I believe I did this way because of the original discussion that we don't want to create a new end point, keep the existing and just change the GeneralSearch to accomodate this keyword search.
When we search inside the model there is really no field permission, role policies check going on there. All of the records will come and only when the query builder runs it make sure it returns only authorized records and skip any unauthorized.Also it is taking care of user's set pagination , so I don't have to deal with that separately.
When search have its own end point this can be optimized further. But at this point, I wouldn't worry about that.